functions/go/set-image/image_transformer_test.go

```diff
@@ -344,9 +352,9 @@ items:
       name: test-container2
 `,
 			ExpectedResults: `
-[info] v1/Pod/the-namespace/the-pod spec.containers.image: set image from foo:1.2.3 to bar:4.5.6
```

Comment:
We return less information now. Why?

Reply:
We should have `ParseResourceList` parse `results` from the input []byte, right now it only parses `items` and `functionConfig`. So I have to use KubeObject and pull the result message one by one. See `runImageTransformerResults`

Once the `ParseResourceList` attaches `results` to `ResourceList`, we can have the unittests include more complete info. 

qq: I am in the impression we are more leveraging the e2e test but not unittests to cover the functions. Or is that only for kpt CLI?